server/authorization.go

```diff
@@ -535,10 +617,16 @@ func (a *AuthorizationServer) quotaExceeded(req *authv3.CheckRequest, envRequest
 func (a *AuthorizationServer) createConditionalEnvoyDenied(
 	req *authv3.CheckRequest, envRequest *config.EnvironmentSpecRequest,
 	tracker *prometheusRequestMetricTracker, authContext *auth.Context,
-	api string, code rpc.Code) *authv3.CheckResponse {
+	api string, err error) *authv3.CheckResponse {
+
+	adapterFault, ok := err.(*fault.AdapterFault)
+	if !ok {
```

Comment:
This branch doesn't seem to have any tests that cover it. Can you please add one?

Reply:
This is currently not reachable, its just a safety check we have put in place.